File: project/.sdk/tm/php/core/Result.php
<?php
declare(strict_types=1);

// ProjectName SDK result

class ProjectNameResult
{
    public bool $ok;
    public int $status;
    public string $status_text;
    public array $headers;
    public mixed $body;
    public mixed $err;
    public mixed $resdata;
    public ?array $resmatch;

    public function __construct(array $resmap = [])
    {
        $this->ok = \Voxgig\Struct\Struct::getprop($resmap, 'ok') === true;
        $s = \Voxgig\Struct\Struct::getprop($resmap, 'status');
        $this->status = is_numeric($s) ? (int)$s : -1;
        $st = \Voxgig\Struct\Struct::getprop($resmap, 'statusText');
        $this->status_text = is_string($st) ? $st : '';
        $h = \Voxgig\Struct\Struct::getprop($resmap, 'headers');
        $this->headers = is_array($h) ? $h : [];
        $this->body = \Voxgig\Struct\Struct::getprop($resmap, 'body');
        $this->err = \Voxgig\Struct\Struct::getprop($resmap, 'err');
        $this->resdata = \Voxgig\Struct\Struct::getprop($resmap, 'resdata');
        $rm = \Voxgig\Struct\Struct::getprop($resmap, 'resmatch');
        $this->resmatch = is_array($rm) ? $rm : null;
    }
}

File: project/.sdk/tm/lua/core/result.lua
-- ProjectName SDK result

local vs = require("utility.struct.struct")

local Result = {}
Result.__index = Result


function Result.new(resmap)
  local self = setmetatable({}, Result)
  resmap = resmap or {}

  self.ok = false
  if vs.getprop(resmap, "ok") == true then
    self.ok = true
  end

  self.status = -1
  local s = vs.getprop(resmap, "status")
  if s ~= nil then
    if type(s) == "number" then
      self.status = math.floor(s)
    end
  end

  self.status_text = ""
  local st = vs.getprop(resmap, "statusText")
  if type(st) == "string" then
    self.status_text = st
  end

  self.headers = {}
  local h = vs.getprop(resmap, "headers")
  if type(h) == "table" then
    self.headers = h
  end

  self.body = vs.getprop(resmap, "body")

  self.err = nil
  local e = vs.getprop(resmap, "err")
  if e ~= nil then
    self.err = e
  end

  self.resdata = vs.getprop(resmap, "resdata")

  self.resmatch = nil
  local rm = vs.getprop(resmap, "resmatch")
  if type(rm) == "table" then
    self.resmatch = rm
  end

  return self
end


return Result
